Jint/Runtime/Modules/ModuleLoader.cs

```diff
@@ -0,0 +1,75 @@
+﻿#nullable enable
+
+using System;
+using System.IO;
+using Esprima;
+using Esprima.Ast;
+
+namespace Jint.Runtime.Modules;
+
+public class ModuleLoader : IModuleLoader
+{
+    private readonly Uri _defaultReferencingLocation;
```

Comment:
Why Uris since it only deals with files? Calling `LocalPath` should fold everything and then we can check for the basepath.